src/rules/padding.ts

```diff
@@ -0,0 +1,438 @@
+/**
+ * @fileoverview Rule to require or disallow newlines between jest functions,
+ *   based on eslint/padding-line-between-statements by Toru Nagashima
+ * @author Ben Kimpel
+ */
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const astUtils = require('eslint/lib/util/ast-utils');
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+// TODO: Check how this behaves when run alongside padding-line-between-statements
+// TODO: Allow customizing pattern to identify test files (can jest provide it? can we detect it with import?)
+
+const LT = `[${Array.from(astUtils.LINEBREAKS).join('')}]`;
+const PADDING_LINE_SEQUENCE = new RegExp(
+  String.raw`^(\s*?${LT})\s*${LT}(\s*;?)$`,
+  'u',
+);
+
+/**
+ * Creates tester which check if an expression node has a certain name
+ *
+ * @param {string} name The Jest function name to test.
+ * @returns {Object} the created tester.
+ * @private
+ */
+function newJestTokenTester(name) {
```

Comment:
This makes a tester object that looks for expression statements with jest tokens. It's based on the patterns from `line-padding-before-statement` which created tester object like this for matching nodes.